### ISSUE TITLE
Allow to change layout of mail for one mail (not global).

### ISSUE THREAD

**Issue by Alex-Bond** [yiisoft/yii2#8084]
For example:

```
Yii::$app->mailer->compose( "@mails/coolMail", [ 'model' => $model ], ['htmlLayout'=>'@mails/layout/anotherLayout'] )
***
                         ->send();
```

Because creating new component instance for each layout is not good.

**Comment by trntv** [comment]
I think you should create this issue here - https://github.com/yiisoft/yii2-swiftmailer/issues

**Comment by ruskid** [comment]
Yep. I also need the possibility to change the layout for a single mail. But there is a better work-around then creating new component for each layout.

You can nest layouts like a normal ones:
```php
<?php $this->beginContent('@common/mail/layouts/single.php') ?>

<?php $this->endContent() ?>
```

**Comment by dynasource** [comment]
issue confirmed

**Comment by gustavovendramini** [comment]
The work-around do not work here, so I fix as follows:

```php
$mailer = Yii::$app->getMailer();
$mailer->htmlLayout = 'layouts/anotherLayout';
```

**Comment by klimov-paul** [comment]
Resolved by #13809
